sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEOptimizer.scala

```diff
@@ -63,3 +70,8 @@ class AQEOptimizer(conf: SQLConf) extends RuleExecutor[LogicalPlan] {
       LogicalPlanIntegrity.checkIfExprIdsAreGloballyUnique(plan))
   }
 }
+
+/**
+ * ConvertToLocalRelation at AQE optimizer side.
+ */
+object AQEConvertToLocalRelation extends ConvertToLocalRelationBase
```

Comment:
I don't get it. There is nothing special in AQE and why not just use `ConvertToLocalRelation`?

Reply:
avoid this rule marked as ineffective during normal optimizer

Reply:
If that's the goal I think we should update the tracking system to measure the effectiveness of AQE optimizer rules separately.

It's super annoying to do this again and again for every normal optimizer rule we add to AQE. e.g. why don't we do it for `UpdateAttributeNullability`?

Reply:
It should also do this. `UpdateAttributeNullability ` updated to `resolveOperatorsUpWithPruning` just now...  see the merged PR [#32686](https://github.com/apache/spark/pull/32686).

Reply:
I was asking why we don't have `AQEUpdateAttributeNullability`.

Reply:
Seems the issue won't happen since we clone the plan step-by-step ( i.e. analyzer/optimizer/physical ) and the ineffective tag won't exist after clone.